rs/index/src/posting_list/file.rs

```diff
@@ -239,6 +227,51 @@ impl FileBackedAppendablePostingListStorage {
             file_offset,
         })
     }
+
+    fn get_slice(
+        mmaps: &[memmap2::MmapMut],
+        start: usize,
+        len: usize,
+        file_size: usize,
+    ) -> Result<&[u8]> {
+        let file_access_info = Self::offset_to_file_access_info(start, file_size, mmaps)?;
+        let (file_offset, file_num) = (file_access_info.file_offset, file_access_info.file_num);
+        let mmap = &mmaps[file_num];
+        Ok(&mmap[file_offset..file_offset + len])
+    }
+
+    fn write_mmaps(
+        writer: &mut BufWriter<&mut File>,
+        mmaps: &[memmap2::MmapMut],
+        offset: usize,
+        backing_file_size: usize,
+    ) -> Result<usize> {
```

Comment:
put the output at the end. change `offset` to `total_length` (which is how I understand this code). Change `backing_file_size` to `individual_backing_file_size`

```suggestion
    fn write_all_mmaps_to_writer(
        mmaps: &[memmap2::MmapMut],
        total_length: usize,
        individual_backing_file_size: usize,
        writer: &mut BufWriter<&mut File>,
    ) -> Result<usize> {
```